### PR TITLE
Preserve stage name when editing scripts with only one stage

### PR DESCRIPTION
**Pull request by davidsbailey**
fixes https://codedotorg.atlassian.net/browse/LP-605

### Background

When editing a script with only one stage, Script#summarize provides the script name in place of the stage name for that stage so that the script name can be displayed instead of the stage name on the script overview page. This breaks the new script editor, which relies on the same Script#summarize method.

This is a similar problem to the one fixed in https://github.com/code-dot-org/code-dot-org/pull/30345, except that Stage#localized_title and Stage#localized_name are a bit more intwined.

### Description

create new `summarize_for_edit` methods on the Script and Stage objects so that we can provide different data to the new script editor than we do to the script overview page. 

I went with a different solution than #30345 here because (1) this code is harder to untangle and (2) we actually want to move in this direction anyway (see Future Work). 

### Future work

 we want to arrive at a situation where we supply only the data needed to allow editing of the stage, and also change the format of the data posted to the server to match that format. splitting off summarize_for_edit like we do in this PR helps get us closer to that situation.

### CODE REVIEW

File: dashboard/test/ui/features/script_overview.feature [hunk 1]
```diff
@@ -29,3 +29,22 @@ Feature: Script overview page
...
     # Make sure we only see student progress, not teacher progress.
     Then I verify progress for stage 29 level 4 is "not_tried"
+
+  Scenario: Script overview contents
+    Given I create a student named "Jean"
+    And I am on "http://studio.code.org/s/allthethings"
+    # make sure we are in summary view and the page has finished loading
+    And I wait until element "td:contains(Maze)" is visible
+    # verify name format in summary view
+    And element "td:contains(2. Maze)" is visible
```
Comment (davidsbailey) [inline dashboard/test/ui/features/script_overview.feature:39]: I wrote these UI tests when I thought I might try to refactor Script#localized_title and localized_name, but decided to keep them in here anyway even though I am no longer touching that codepath.

File: dashboard/app/models/stage.rb [hunk 2]
```diff
@@ -87,11 +87,7 @@ def localized_name
   end
 
   def localized_category
-    if flex_category
-      I18n.t "flex_category.#{flex_category}"
-    else
-      I18n.t "flex_category.content"
-    end
+    flex_category && I18n.t("flex_category.#{flex_category}")
```
Comment (maddiedierker) [inline dashboard/app/models/stage.rb]: it's expected that we no longer want to return `I18n.t "flex_category.content"` here if there is no flex category?

Comment (davidsbailey) [reply dashboard/app/models/stage.rb]: this was a mistake with the merge base -- I was trying to stack this on top of https://github.com/code-dot-org/code-dot-org/pull/30345. But now it looks like there is a problem with the first half of the solution in PR https://github.com/code-dot-org/code-dot-org/pull/30345, so I am just going to rebase this to be directly off of staging.

**Review by maddiedierker — APPROVED** [review]
🎉

**Pull request merged**